File: api/routes/health.py
"""Health check endpoint -- also the phone's post-pair connectivity probe."""

from fastapi import APIRouter, Request

from api.config import settings
from api.utils.platform import machine_platform

router = APIRouter()


@router.get("/api/health")
async def health(request: Request):
    transport = getattr(request.app.state, "transport", None)
    return {
        "status": "healthy",
        "modules": {
            "computer_use": settings.computer_use_enabled,
        },
        "platform": machine_platform(),
        "version": settings.version,
        "transport": transport.status() if transport is not None else None,
    }

File: rust/src/routes/health.rs
use crate::config::VERSION;
use crate::state::AppState;
use axum::Json;
use axum::extract::State;
use serde_json::{Value, json};

/// Unauthenticated: it is the phone's post-pair connectivity probe, so it has
/// to answer before a token exists.
pub async fn health(State(state): State<AppState>) -> Json<Value> {
    let computer_use_installed = state.computer_use_status.read().unwrap()["venv_ready"]
        .as_bool()
        .unwrap_or(false);
    Json(json!({
        "status": "healthy",
        "modules": { "computer_use": computer_use_installed },
        "platform": crate::platform::machine_platform(),
        "version": VERSION,
        "transport": state.transport.status(),
    }))
}
